### PR TITLE
Stop using square icons for SOCIAL and update FA to 4.6.3

### PR DESCRIPTION
**Pull request by xenithorb**
There are a very limited number of square icons with FA and for easier support with other icons like Gitlab (the one was targeting the most that has no -square) I removed the loop that inserts square into the class. My personal take is that these icons look more uniform, anyhow. 

I also upgraded FA to version 4.6.3 because this is the newest version and also includes the Gitlab icon, as well as many newer brand icons, too.

### CODE REVIEW

**Comment by ingwinlu** [comment]
hi,

the problem when not using `-squared` is that the icons seem to vary in width:
capture

So we would need to add `fa-fw` to fix the width of the icons resulting in 
capture

Otherwise I see no problem in removing `-squared` icons.

**Comment by xenithorb** [comment]
Ah, very nice. I had noticed that but I'm not used to working with FA, so I wasn't aware it had that class. I'll make the change ASAP

File: templates/base.html [hunk 1]
```diff
@@ -18,11 +18,7 @@
...
 {% macro get_icon_attributes(name) -%}
     {%- set name_sanitized = name|lower|replace('+','-plus')|replace(' ','-') -%}
-    {%- if name_sanitized in ['flickr', 'spotify', 'stack-overflow'] -%}
-        {%- set iconattributes = 'fa fa-' ~ name_sanitized ~ ' fa-lg' -%}
-    {%- else -%}
-        {%- set iconattributes = 'fa fa-' ~ name_sanitized ~ '-square fa-lg' -%}
-    {%- endif -%}
+    {%- set iconattributes = 'fa fa-' ~ name_sanitized ~ ' fa-lg fa-fw' -%}
     {%- if name|lower == 'email' -%}
         {%- set iconattributes = 'fa fa-envelope-square fa-lg' -%}
```
Comment (ingwinlu) [inline templates/base.html]: Do you think this should be without -square then as well? I missed it when looking through the changes the first time.

Comment (xenithorb) [inline templates/base.html]: You're talking about these:

```
templates/base.html:23:        {%- set iconattributes = 'fa fa-envelope-square fa-lg' -%}
templates/base.html:100:                            <i class="fa fa-facebook-square padding-small"></i>
templates/base.html:110:                            <i class="fa fa-twitter-square padding-small"></i>
templates/base.html:188:                            <i class="fa fa-facebook-square padding-small"></i>
templates/base.html:200:                            <i class="fa fa-twitter-square padding-small"></i>
```
right?

**Comment by xenithorb** [comment]
0ec27c2 seems to work, my only concern is that, _while the share icons are aligned_ there's an illusion that they're not because the facebook icon's f is off-center:

image

While the SOCIAL icons look just fine. 

I'm personally ok with it this way... but maybe you have other thoughts..

**Comment by ingwinlu** [comment]
Maybe we can use something like http://fontawesome.io/examples/#stacked to counteract that behaviour. I.e. drop -squared because of the reasons you listed, but still keeping the 'uniform' width effect that it provided. Might be the best of both worlds though I still have to try it.

**Comment by xenithorb** [comment]
I'm trying to test this out but I'm apparently totally confused about how modern webdesign works. I inserted the following into `base.html` and was perplexed for a while about why the site wasn't updating even after `devserver` said it was regenerating the page.  I was confused because on inspecting the element the new classes and additional icon element were completely missing (not using cache, private browsing windows, chrome and firefox). It wasn't until I actually "view source"d that I could see that it was there in the source of the page but the browser was ignoring everything I added for some reason. I'm totally lost as to why. 

```html
    <li class="fa-stack fa-lg">
        <a href="{{ link }}" title="{{name}}">
          <i class="fa fa-stack-2x fa-square-o"></i>
          <i class="{{get_icon_attributes(name|striptags|trim)}} fa-stack-1x"></i>
        </a>
   </li>
```

**Comment by ingwinlu** [comment]
Update to bootstrap 4 + fontawesome changes as well so closing for now

**Pull request closed without merging**